impeller/tessellator/tessellator.cc

```diff
@@ -128,6 +128,77 @@ Tessellator::Result Tessellator::Tessellate(
   return Result::kSuccess;
 }
 
+Tessellator::Result Tessellator::TessellateBuilder(
+    FillType fill_type,
+    const Path::Polyline& polyline,
+    const BuilderCallback& callback) const {
+  if (!callback) {
+    return Result::kInputError;
+  }
+
+  if (polyline.points.empty()) {
+    return Result::kInputError;
+  }
+
+  auto tessellator = c_tessellator_.get();
+  if (!tessellator) {
+    return Result::kTessellationError;
+  }
+
+  constexpr int kVertexSize = 2;
+  constexpr int kPolygonSize = 3;
+
+  //----------------------------------------------------------------------------
+  /// Feed contour information to the tessellator.
+  ///
+  static_assert(sizeof(Point) == 2 * sizeof(float));
+  for (size_t contour_i = 0; contour_i < polyline.contours.size();
+       contour_i++) {
+    size_t start_point_index, end_point_index;
+    std::tie(start_point_index, end_point_index) =
+        polyline.GetContourPointBounds(contour_i);
+
+    ::tessAddContour(tessellator,  // the C tessellator
+                     kVertexSize,  //
+                     polyline.points.data() + start_point_index,  //
+                     sizeof(Point),                               //
+                     end_point_index - start_point_index          //
+    );
+  }
+
+  //----------------------------------------------------------------------------
+  /// Let's tessellate.
+  ///
+  auto result = ::tessTesselate(tessellator,                   // tessellator
+                                ToTessWindingRule(fill_type),  // winding
+                                TESS_POLYGONS,                 // element type
+                                kPolygonSize,                  // polygon size
+                                kVertexSize,                   // vertex size
+                                nullptr  // normal (null is automatic)
+  );
+
+  if (result != 1) {
+    return Result::kTessellationError;
+  }
+
+  int vertexItemCount = tessGetVertexCount(tessellator) * kVertexSize;
+  auto vertices = tessGetVertices(tessellator);
+  int elementItemCount = tessGetElementCount(tessellator) * kPolygonSize;
+  auto elements = tessGetElements(tessellator);
+  // libtess uses an int index internally due to usage of -1 as a sentinel
+  // value.
+  std::vector<uint16_t> indices;
+  indices.reserve(elementItemCount);
+  for (int i = 0; i < elementItemCount; i++) {
+    indices.emplace_back((uint16_t)elements[i]);
```

Comment:
```suggestion
    indices.emplace_back(static_cast<uint16_t>(elements[i]));
```

Reply:
I think this needs to be reinterpret_cast right?

Reply:
nope, static_cast is right...

Reply:
Done